src/providers/krb5/krb5_child.c

```diff
@@ -3154,6 +3158,49 @@ static int k5c_setup(struct krb5_req *kr, uint32_t offline)
     return kerr;
 }
 
+static krb5_error_code validate_keytab(struct krb5_req *kr)
+{
+    krb5_error_code kerr;
+    char krb5_conf_keytab[MAX_KEYTAB_NAME_LEN];
+    char *path_start = NULL;
+
+    if (kr->keytab == NULL && (
+        kr->pd->cmd == SSS_PAM_AUTHENTICATE ||
+        kr->pd->cmd == SSS_PAM_PREAUTH ||
+        kr->pd->cmd == SSS_CMD_RENEW ||
+        kr->pd->cmd == SSS_PAM_CHAUTHTOK_PRELIM ||
+        kr->pd->cmd == SSS_PAM_CHAUTHTOK)) {
+
+        DEBUG(SSSDBG_TRACE_FUNC,
+              "Missing krb5_keytab option for domain, looking for default one\n");
+
+        kerr = krb5_kt_default_name(kr->ctx, krb5_conf_keytab, sizeof(krb5_conf_keytab));
+        if (kerr != 0) {
+            DEBUG(SSSDBG_CRIT_FAILURE,
+                  "Unable to get default keytab location from krb.conf\n");
+            return kerr;
+        }
+
+        DEBUG(SSSDBG_TRACE_FUNC,
+              "krb5_kt_default_name() returned: %s\n",
+              krb5_conf_keytab);
+
+        /* krb5_kt_default_name() can return file path with "FILE:" prefix,
+           it need to be removed */
+        if (0 == strncmp(krb5_conf_keytab, "FILE:", strlen("FILE:"))) {
```

Comment:
In krb5_child it is not needed to remove the `FILE:` prefix because if I'm not mistaken `kr->keytab` is only used in debug messages or with `krb5_kt_resolve()` which can handle the prefix.

Reply:
I checked this and you are right. But I think that it may be safer to remove `FILE:` prefix from `kr->keytab`. For person not familiar with the kerberos it may not be obvious that this prefix may be there. As a developer I would expect (judging by the variable name) that `kr->keytab` is always valid filesystem path (which is not true with `FILE:` prefix).